tests/Unit/Domain/Test_ElementId.cpp

```diff
@@ -117,9 +122,51 @@ void test_element_id() {
   CHECK(ElementId<3>::external_boundary_id().segment_ids() ==
         make_array<3>(SegmentId(SegmentId::max_refinement_level, 0)));
 }
+
+template <size_t VolumeDim>
+void test_serialization() noexcept {
```

Comment:
This test is more complex than I expect. For my understanding: why is the work repeated for every possible value of block_id?

Reply:
I was testing the domain @wthrowe created that was having strange behavior; it is how I found the missing assert in SegmentId, so I figured I would keep it...